### PR TITLE
feat: countdown to prize-show end on upcoming giveaway banner

### PR DESCRIPTION
**Pull request by briankeane**
The player-page upcoming-giveaway banner now leads with "[Prize] giveaway in the next ~40 min" while the prize show is the one currently on air, derived from `spin.airing.endTime` (a new optional field added in PlayolaPlayer 0.19.1) matched to the prize show by `airingId`. This bounds the window without revealing the exact open moment, and falls back to the prior timeless "Win a [Prize] — coming up on [Station]" copy when the show isn't airing, the end time is unknown, or the window has passed. The phrase refreshes once a minute, rounds to nearest 5 min with a "few minutes" floor near show-end. Visibility rules are unchanged — the banner still only shows before the tap button and never reappears after. Adds 6 model tests covering the countdown, rounding, and fallback paths.

Note: requires PlayolaPlayer ≥ 0.19.1 (already satisfied by the repo's `upToNextMinor 0.19.0` pin); the same `Airing.startTime/endTime` change still needs to land on the 0.20.0 line.

## Summary by CodeRabbit

* **New Features**
  * Giveaway banners now display a timed “in the next ~X min” message (rounded to 5-minute increments) when an upcoming prize window can be determined.
  * The countdown updates automatically while the banner is active.

* **Bug Fixes**
  * Banner text now falls back to the existing timeless message when the timing window is unavailable, missing, or already passed, including when the airing doesn’t match.

* **Tests**
  * Expanded and refactored banner visibility and message-generation tests to cover timing and matching edge cases.

### CODE REVIEW

File: PlayolaRadio/Views/Pages/PlayerPage/UpcomingGiveawayBannerModel.swift [hunk 1]
```diff
@@ -7,16 +7,38 @@ import Sharing
...
+  // MARK: - Properties
+
+  /// Current time, refreshed once a minute by `task()`. `nil` until the view starts the refresh; the
+  /// window phrase stays hidden (timeless fallback) until then.
+  var now: Date?
+
   // MARK: - Initialization
   override init() {
     super.init()
   }
```
[the remark this answers is not in the document]
Comment (briankeane) [reply PlayolaRadio/Views/Pages/PlayerPage/UpcomingGiveawayBannerModel.swift]: Fixed — moved Initialization before Properties to match the required section order.

File: PlayolaRadio/Views/Pages/PlayerPage/UpcomingGiveawayBannerModelTests.swift [hunk 2]
```diff
@@ -9,24 +10,49 @@ import Testing
...
+        id: id, stationId: station, prizeName: prize, winningNumber: 9, status: .scheduled,
+        airingId: airingId))
+  }
+
+  /// Builds the model and simulates the once-a-minute tick having fired at `referenceNow`.
+  private func makeModel() -> UpcomingGiveawayBannerModel {
+    let model = UpcomingGiveawayBannerModel()
+    model.now = Self.referenceNow
+    return model
   }
```
[the remark this answers is not in the document]
Comment (briankeane) [reply PlayolaRadio/Views/Pages/PlayerPage/UpcomingGiveawayBannerModelTests.swift:49]: Added a test for the now==nil path (usesCurrentDateForWindowBeforeFirstTick). Note the behavior changed per CodeRabbit's adjacent comment: windowPhrase now falls back to currentDate when now is nil, so first paint shows the countdown rather than the timeless copy. The new test asserts that.

File: PlayolaRadio/Views/Pages/PlayerPage/UpcomingGiveawayBannerModel.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (briankeane) [reply PlayolaRadio/Views/Pages/PlayerPage/UpcomingGiveawayBannerModel.swift]: Fixed — reordered to Initialization before Properties.

File: PlayolaRadio/Views/Pages/PlayerPage/UpcomingGiveawayBannerModel.swift [hunk 3]
```diff
@@ -7,16 +7,38 @@ import Sharing
...
   // MARK: - Shared State
   @ObservationIgnored @Shared(.nowPlaying) var nowPlaying
   @ObservationIgnored @Shared(.activeGiveaway) var activeGiveaway
   @ObservationIgnored @Shared(.upcomingGiveaways) var upcomingGiveaways
 
+  // MARK: - Properties
+
+  /// Current time, refreshed once a minute by `task()`. `nil` until the view starts the refresh; the
+  /// window phrase stays hidden (timeless fallback) until then.
+  var now: Date?
```
[the remark this answers is not in the document]
Comment (briankeane) [reply PlayolaRadio/Views/Pages/PlayerPage/UpcomingGiveawayBannerModel.swift]: Fixed — windowPhrase now uses now ?? currentDate, so a currently airing prize show renders the countdown on first paint. The date dependency is read lazily only while a countdown renders, so constructing the model elsewhere (e.g. PlayerPageTests) doesn't touch it.
